.travis.yml

```diff
@@ -0,0 +1,50 @@
+language: csharp
+
+solution: GPUVerify.sln
+
+env:
+  global:
+    - LLVM_VERSION=5.0
+    - LLVM_CONFIG="llvm-config-${LLVM_VERSION}"
+    - Z3_VERSION=4.5.0
+    - CVC4_VERSION=1.5
+    - CLONE_DEPTH=50
+
+cache:
+  directories:
+    - $TRAVIS_BUILD_DIR/downloads
+
+addons:
+  apt:
+    sources:
+      - llvm-toolchain-trusty-5.0
+    packages:
+      - g++
+      - ninja-build
+      - python-psutil
+      # From sources above
+      - llvm-5.0-dev
+      - clang-5.0
+      - clang-5.0-dev
+
+matrix:
+  include:
+    - os: linux
+      dist: trusty
+      sudo: false
+      env:
+        - DEFAULT_SOLVER=z3
+        - Z3="z3-${Z3_VERSION}-x64-ubuntu-14.04"
+    - os: linux
+      dist: trusty
+      sudo: false
+      env:
+        - DEFAULT_SOLVER=cvc4
+        - CVC4="cvc4-${CVC4_VERSION}-x86_64-linux-opt"
+
+script:
+  - export GPUVERIFY_DIR=${TRAVIS_BUILD_DIR}
+  - export BUGLE_DIR=${TRAVIS_BUILD_DIR}/bugle
+  - export DOWNLOADS_DIR=${TRAVIS_BUILD_DIR}/downloads
+  - git clone --depth=${CLONE_DEPTH} --branch=master https://github.com/jeroenk/bugle.git
```

Comment:
Why is this step here and not in `build_and_test.sh`?

Reply:
Because of the incredibly nasty thing I do here: https://github.com/mc-imperial/bugle/pull/1